unicodetools/data/ucd/dev/DerivedNormalizationProps.txt

```diff
@@ -9005,6 +9005,6078 @@ E01F0..E0FFF  ; NFKC_CF;                # Cn [3600] <reserved-E01F0>..<reserved-
+00D6          ; NFKC_SCF; 00F6           # L&       LATIN CAPITAL LETTER O WITH DIAERESIS
+00D8          ; NFKC_SCF; 00F8           # L&       LATIN CAPITAL LETTER O WITH STROKE
+00D9          ; NFKC_SCF; 00F9           # L&       LATIN CAPITAL LETTER U WITH GRAVE
+00DA          ; NFKC_SCF; 00FA           # L&       LATIN CAPITAL LETTER U WITH ACUTE
+00DB          ; NFKC_SCF; 00FB           # L&       LATIN CAPITAL LETTER U WITH CIRCUMFLEX
+00DC          ; NFKC_SCF; 00FC           # L&       LATIN CAPITAL LETTER U WITH DIAERESIS
+00DD          ; NFKC_SCF; 00FD           # L&       LATIN CAPITAL LETTER Y WITH ACUTE
+00DE          ; NFKC_SCF; 00FE           # L&       LATIN CAPITAL LETTER THORN
+0100          ; NFKC_SCF; 0101           # L&       LATIN CAPITAL LETTER A WITH MACRON
+0102          ; NFKC_SCF; 0103           # L&       LATIN CAPITAL LETTER A WITH BREVE
+0104          ; NFKC_SCF; 0105           # L&       LATIN CAPITAL LETTER A WITH OGONEK
+0106          ; NFKC_SCF; 0107           # L&       LATIN CAPITAL LETTER C WITH ACUTE
+0108          ; NFKC_SCF; 0109           # L&       LATIN CAPITAL LETTER C WITH CIRCUMFLEX
+010A          ; NFKC_SCF; 010B           # L&       LATIN CAPITAL LETTER C WITH DOT ABOVE
+010C          ; NFKC_SCF; 010D           # L&       LATIN CAPITAL LETTER C WITH CARON
+010E          ; NFKC_SCF; 010F           # L&       LATIN CAPITAL LETTER D WITH CARON
+0110          ; NFKC_SCF; 0111           # L&       LATIN CAPITAL LETTER D WITH STROKE
+0112          ; NFKC_SCF; 0113           # L&       LATIN CAPITAL LETTER E WITH MACRON
+0114          ; NFKC_SCF; 0115           # L&       LATIN CAPITAL LETTER E WITH BREVE
+0116          ; NFKC_SCF; 0117           # L&       LATIN CAPITAL LETTER E WITH DOT ABOVE
+0118          ; NFKC_SCF; 0119           # L&       LATIN CAPITAL LETTER E WITH OGONEK
+011A          ; NFKC_SCF; 011B           # L&       LATIN CAPITAL LETTER E WITH CARON
+011C          ; NFKC_SCF; 011D           # L&       LATIN CAPITAL LETTER G WITH CIRCUMFLEX
+011E          ; NFKC_SCF; 011F           # L&       LATIN CAPITAL LETTER G WITH BREVE
+0120          ; NFKC_SCF; 0121           # L&       LATIN CAPITAL LETTER G WITH DOT ABOVE
+0122          ; NFKC_SCF; 0123           # L&       LATIN CAPITAL LETTER G WITH CEDILLA
+0124          ; NFKC_SCF; 0125           # L&       LATIN CAPITAL LETTER H WITH CIRCUMFLEX
+0126          ; NFKC_SCF; 0127           # L&       LATIN CAPITAL LETTER H WITH STROKE
+0128          ; NFKC_SCF; 0129           # L&       LATIN CAPITAL LETTER I WITH TILDE
+012A          ; NFKC_SCF; 012B           # L&       LATIN CAPITAL LETTER I WITH MACRON
+012C          ; NFKC_SCF; 012D           # L&       LATIN CAPITAL LETTER I WITH BREVE
+012E          ; NFKC_SCF; 012F           # L&       LATIN CAPITAL LETTER I WITH OGONEK
+0130          ; NFKC_SCF; 0069           # L&       LATIN CAPITAL LETTER I WITH DOT ABOVE
```

Comment:
This is weird; I would expect it to NFKC_SCF to itself, since it SCFs to itself.

I guess this has something to do with it:
https://github.com/unicode-org/unicodetools/blob/82df28717a1075c227dfcdd07d889c7b1f6a038a/unicodetools/src/main/java/org/unicode/text/UCD/UCD.java#L630-L638

@macchiati, do you have context on this special case? Where does this `"I"` come from?

Reply:
OK, this if statement is a red herring, and quite possibly dead code.

`udata.simpleCaseFolding` is `i` for U+0130.
Which is wrong (U+0130 is invariant under SCF, see UnicodeData.txt. or the JSPs, or anything that talks about the simple case folding of that character).

Reply:
OK, so, trying to piece things together.
* NFKC_Casefold (and now NFKC_Simple_Casefold) is based on case folding with `getCase`, which uses the case folding fields in `UData`, which ultimately comes from UnicodeData.txt, which does not have the case foldings; instead those are derived from the case mappings with hardcoded exceptions for, e.g., Cherokee; but since there is no such exception for U+0130, it gets incorrectly simple-casefolded to U+0069, its lowercase mapping: https://github.com/unicode-org/unicodetools/blob/70dce2c89f185c65b436c28404ae5b7bdb32c2d1/unicodetools/src/main/java/org/unicode/text/UCD/UData.java#L225-L239.
* CaseFolding.txt, the actual normative source of the foldings, is a generated file; it is generated by `GenerateCaseFolding.java`, which only depends on `getCase` for upper and lower case _mappings_, not case folding; it has its own set of hardcoded exceptions, which include U+0130 as well as Cherokee.

Having two independent definitions of case folding is confusing, but the invariants tests will prevent us from violating the stability policy, right?
https://github.com/unicode-org/unicodetools/blob/0cf2dd2539e79e61a4ea792e83af11d923362572/unicodetools/src/main/resources/org/unicode/text/UCD/UnicodeInvariantTest.txt#L171-L175

😬

Reply:
I think you are right about U+0130, but looping in @markusicu because we spent hours working over the normalization properties of the 4 i's, and we need to triple check this!

Reply:
Yes, I think this has been resolved by #427; please check the analysis in https://github.com/unicode-org/unicodetools/pull/423#issuecomment-1491274525.